frontend/src/pages/modelServing/screens/global/useRouteForInferenceService.ts

```diff
@@ -14,34 +11,40 @@ const useRouteForInferenceService = (
   const [loaded, setLoaded] = React.useState(false);
   const [loadError, setLoadError] = React.useState<Error | null>(null);
 
-  const routeName = inferenceService.metadata.name;
+  const routeName = isKServe
+    ? `${inferenceService.metadata.name}-predictor`
+    : inferenceService.metadata.name;
   const routeNamespace = inferenceService.metadata.namespace;
-  const kserveRoute = isKServe ? getUrlFromKserveInferenceService(inferenceService) : null;
-  const state = getInferenceServiceActiveModelState(inferenceService);
-  const kserveLoaded = state === InferenceServiceModelState.LOADED;
 
   React.useEffect(() => {
     if (isKServe) {
-      setRoute(kserveRoute || null);
-      setLoaded(kserveLoaded);
-      setLoadError(kserveLoaded ? (kserveRoute ? null : new Error('Route not found')) : null);
-      return;
+      getKNativeService(routeName, routeNamespace)
+        .then((route) => {
+          setRoute(route.status.url);
+          setLoaded(true);
+        })
+        .catch((e) => {
+          setLoadError(e);
+          setLoaded(true);
+        });
     }
+
     if (!isRouteEnabled) {
       setLoadError(null);
       setLoaded(true);
       return;
     }
+
     getRoute(routeName, routeNamespace)
       .then((route) => {
-        setRoute(`https://${route.spec.host}${route.spec.path}`);
+        setRoute(`https://${route.spec.host}${route.spec.path}/infer`);
```

Comment:
So, for KServe here, it won't be /infer, it'll be one of a few options the user could pick depending on the runtime since we have to use them in passthru mode

Reply:
yeah, that was for modelmesh, thanks!